torchgeo/trainers/segmentation.py

```diff
@@ -232,7 +315,10 @@ def training_step(
         loss: Tensor = self.criterion(y_hat, y)
         self.log('train_loss', loss, batch_size=batch_size)
         self.train_metrics(y_hat, y)
-        self.log_dict(self.train_metrics, batch_size=batch_size)
+        self.log_dict(
+            {f'{k}': v for k, v in self.train_metrics.compute().items()},
```

Comment:
I think that ``compute()`` should not be called per step, since this will cause a reduction to happen multiple times per epoch and in the end the output is the average of the average of the average...

Reply:
Can be addressed with `on_epoch=True`?

Reply:
Not sure exactly how this works, but I found it gave wrong results for when ``on_epoch`` wasn't specified.